winml/adapter/winml_adapter_model.h

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+#pragma once
+
+#include "winml_adapter_c_api.h"
+#include <memory>
+#include "core/graph/onnx_protobuf.h"
+
+class ModelInfo;
+
+struct OrtModel {
+ public:
+  static OrtStatus* CreateOrtModelFromPath(const char* path, size_t len, OrtModel** model);
+  static OrtStatus* CreateOrtModelFromData(void* data, size_t len, OrtModel** model);
+  static OrtStatus* CreateOrtModelFromProto(std::unique_ptr<onnx::ModelProto>&& model_proto, OrtModel** model);
+  const ModelInfo* UseModelInfo() const;
+
+  const onnx::ModelProto* UseModelProto() const;
+  std::unique_ptr<onnx::ModelProto> DetachModelProto();
+
+ private:
+  OrtModel(std::unique_ptr<onnx::ModelProto>&& model_proto);
```

Comment:
why && ?

Reply:
oops i should remove that.

For unique_ptr, in reality it does nothing, as unique_ptr itself disables copy construction from const lvalues&.
For custom types, which support copy AND move construciton, the rvalue reference here would prevent callers from passing in lvalues.

I will remove for unique_ptr as it already handles this.

---
In reply to: [370826302](https://github.com/microsoft/onnxruntime/pull/2891#discussion_r370826302) [](ancestors = 370826302)